src/Payum/Action/AssertAction.php

```diff
@@ -7,15 +7,20 @@
 use CommerceWeavers\SyliusSaferpayPlugin\Client\SaferpayClientInterface;
 use CommerceWeavers\SyliusSaferpayPlugin\Client\ValueObject\AssertResponse;
 use CommerceWeavers\SyliusSaferpayPlugin\Client\ValueObject\ErrorResponse;
+use CommerceWeavers\SyliusSaferpayPlugin\Payum\Action\Assert\FailedResponseHandlerInterface;
+use CommerceWeavers\SyliusSaferpayPlugin\Payum\Action\Assert\SuccessfulResponseHandlerInterface;
 use CommerceWeavers\SyliusSaferpayPlugin\Payum\Request\Assert;
 use Payum\Core\Action\ActionInterface;
 use Payum\Core\Exception\RequestNotSupportedException;
 use Sylius\Component\Core\Model\PaymentInterface;
 
 final class AssertAction implements ActionInterface
 {
-    public function __construct(private SaferpayClientInterface $saferpayClient)
-    {
+    public function __construct(
+        private SaferpayClientInterface $saferpayClient,
+        private SuccessfulResponseHandlerInterface $successfulResponseHandler,
+        private FailedResponseHandlerInterface $failedResponseHandler,
```

Comment:
Does it make sense to introduce service locator based on response? /cc @Zales0123

Reply:
Maybe, but probably not needed for now for only 2 handlers 🖖